File: BotTemplate/BotTemplate-Java/src/main/java/io/github/freya022/bot/resolvers/TimeUnitResolver.java
package io.github.freya022.bot.resolvers;

import com.freya02.botcommands.api.core.service.annotations.Resolver;
import com.freya02.botcommands.api.parameters.ParameterResolver;
import com.freya02.botcommands.api.parameters.Resolvers;

import java.util.concurrent.TimeUnit;

@SuppressWarnings("unused")
public class TimeUnitResolver {
    @Resolver
    public static ParameterResolver<?, ?> get() {
        return Resolvers.enumResolver(
                TimeUnit.class,
                new TimeUnit[]{TimeUnit.SECONDS, TimeUnit.MINUTES, TimeUnit.HOURS, TimeUnit.DAYS}
        );
    }
}

File: BotTemplate/BotTemplate-Kotlin/src/main/kotlin/io/github/freya022/bot/resolvers/TimeUnitResolver.kt
package io.github.freya022.bot.resolvers

import com.freya02.botcommands.api.core.service.annotations.Resolver
import com.freya02.botcommands.api.localization.context.LocalizationContext
import com.freya02.botcommands.api.localization.to
import com.freya02.botcommands.api.parameters.enumResolver
import java.util.concurrent.TimeUnit

@Suppress("unused")
object TimeUnitResolver {
    // The displayed name should be lowercase with the first letter uppercase, see Resolvers#toHumanName
    @Resolver
    fun get() = enumResolver<TimeUnit>(TimeUnit.SECONDS, TimeUnit.MINUTES, TimeUnit.HOURS, TimeUnit.DAYS)
}

fun TimeUnit.localize(time: Long, localizationContext: LocalizationContext): String {
    return localizationContext.switchBundle("Misc")
        .localizeOrNull("time_unit.$name", "time" to time)
        ?: name.lowercase().trimEnd('s')
}